### DISCUSSION TITLE
Cache API in Coil 2

### DISCUSSION THREAD

**Discussion by yschimke** [coil-kt/coil#1105] [General]
Just wondering about the Cache API in Coil 2.  It's implemented internally via Okio Filesystem, wondering whether it's useful to use that as the API also instead of java.io.File?

```
interface DiskCache {
    /** The directory where the cache stores its data. */
    val directory: File
```    

Not critical, just easier when I'm using Okio on my side anyway, and for testing via Okio FakeFileSystem etc.

**Comment by colinrtwhite** [comment]
@yschimke I think this is a great idea! Open to a PR if you'd like to implement this otherwise I can give it a shot. I think we'd probably want to convert all `File` references in `DiskCache` and `DiskLruCache` to `Path`s. As you mentioned, we should also expose `DiskCache.Builder.fileSystem`.

**Comment by yschimke** [comment]
Happy to.

**Reply by yschimke** [reply]
https://github.com/coil-kt/coil/pull/1115
